src/nat/runtime/session.py

```
@@ -139,14 +460,39 @@ async def run(self, message, runtime_type: RuntimeTypeEnum = RuntimeTypeEnum.RUN
+                    await asyncio.wait_for(self._per_user_builders_cleanup_task, timeout=5.0)
+                except TimeoutError:
+                    logger.warning("Cleanup task did not finish in time, cancelling")
+                    self._per_user_builders_cleanup_task.cancel()
+
+            # Cleanup all per-user builders
+            async with self._per_user_builders_lock:
+                for user_id, builder_info in list(self._per_user_builders.items()):
+                    logger.debug(f"Cleaning up per-user builder for user {user_id}")
+                    try:
+                        await builder_info.builder.__aexit__(None, None, None)
+                    except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)